front/src/applications/editor/tools/routeEdition/components/RouteEditionPanel.tsx

```diff
@@ -0,0 +1,314 @@
+        ...getRouteEditionState({
+          ...entity,
+          properties: result && result[0] ? result[0].railjson : entity.properties,
+        } as RouteEntity),
+        // we keep the extremetiesEntity state, it has not changed
+        extremitiesEntity: prev.extremitiesEntity,
+      }));
+    },
+    [infraID, dispatch, setState]
+  );
+
+  /**
+   * What we do when the rigid transit checkbox changed
+   */
+  const onReleaseDetectorChange = useCallback(
+    (withDetector: boolean) => {
+      setState((prev) => {
+        let detectors: string[] = [];
+        if (
+          withDetector &&
+          prev.optionsState.type === 'options' &&
+          !isNil(prev.optionsState.focusedOptionIndex)
+        ) {
+          const selected = prev.optionsState.options[prev.optionsState.focusedOptionIndex];
+          detectors = selected.data.detectors;
+        }
+        return {
+          ...prev,
+          entity: {
+            ...prev.entity,
+            properties: {
+              ...prev.entity.properties,
+              release_detectors: detectors,
+            },
+          },
+        };
+      });
+    },
+    [setState]
+  );
+
+  /**
+   * Function to sezarch for routes
+   */
+  const searchAltRoutes = useCallback(async () => {
+    const route = state.entity;
+    try {
+      setState({
+        optionsState: { type: 'loading' },
+      });
+      if (!infraID) throw new Error('No infra selected');
+      const payload = await getCompatibleRoutesPayload(
+        infraID,
+        route.properties.entry_point,
+        route.properties.exit_point,
+        dispatch
+      );
+      const candidates = await postPathfinding({
+        infraId: infraID,
+        pathfindingInput: payload,
+      }).unwrap();
+
+      const candidateColors = chroma
+        .scale(['#321BF7CC', '#37B5F0CC', '#F0901FCC', '#F7311BCC', '#D124E0CC'])
+        .mode('lch')
+        .colors(candidates.length)
+        .map((str) => chroma(str).hex());
+
+      const features = await getRouteGeometries(
+        infraID as number,
+        route.properties.entry_point,
+        route.properties.exit_point,
+        candidates,
+        dispatch
+      );
+
+      setState({
+        optionsState: {
+          type: 'options',
+          options: candidates.map((candidate, i) => ({
+            data: candidate,
+            color: candidateColors[i],
+            feature: { ...features[i], properties: { color: candidateColors[i], index: i } },
+          })),
+        },
+      });
+    } catch (e) {
+      setState({ optionsState: { type: 'idle' } });
+    }
+  }, [infraID, setState, state.entity]);
+
+  /**
+   * Funciton that select a new route search item
+   */
+  const selectSearchRouteItem = useCallback(
+    (index: number) => {
+      setState((prev) => {
+        if (prev.optionsState.type !== 'options' || !prev.entity) return prev;
+
+        const isSelection = index !== prev.optionsState.focusedOptionIndex;
```

Comment:
I find it a bit weird that isSelection is true if the item is not selected. Here are some suggestions

```suggestion
        const isSelectionAction = index !== prev.optionsState.focusedOptionIndex;
        const isItemUnselected = index !== prev.optionsState.focusedOptionIndex;
        const isItemSelected = index === prev.optionsState.focusedOptionIndex;
```